tokio-quiche/src/quic/router/mod.rs

```diff
@@ -522,11 +547,25 @@ where
                                 },
                                 ControlMessageOwned::Ipv4PacketInfo(_) |
                                 ControlMessageOwned::Ipv6PacketInfo(_) => {
-                                    // We only want the destination address from
-                                    // IP_RECVORIGDSTADDR, but we'll get these
+                                    // We only want the destination address
+                                    // from
+                                    // IP_RECVORIGDSTADDR, but we'll get
+                                    // these
                                     // messages because
                                     // we set IP_PKTINFO on the socket.
                                 },
+                                ControlMessageOwned::Unknown(raw_cmsg) => {
+                                    let UnknownCmsg {
+                                        cmsg_header,
+                                        data_bytes,
+                                    } = raw_cmsg;
+
+                                    if cmsg_header.cmsg_level == SOL_SOCKET &&
```

Comment:
Don't you need some `#[cfg(target_os = "linux")]` protection here?

Reply:
> Don't you need some #[cfg(target_os = "linux")] protection here?

The whole block should already be Linux-protected (see L411)

> Also, consider putting unknown cmsg handling in a function for code clarity, since I guess we might need more of these in future

I think it could be a bit annoying for now because of the scoping issues - are we okay leaving as-is for now and then adding a helper function if/when another cmsg type is required?